### PR TITLE
Move save start to lua

### PR DESCRIPTION
**Pull request by theis999**
**Reversed changes to save and start.**

**Save global**
Now you can just insert a save command, and then load that save. No need for start anymore.

Added lua global tas data, to be persistent between saves
Added methods to migrate data to and from global tas data
Re-added queued save, which queues in the pre-tick and executes in post-tick

Known bug, if you have multiple saves in one pre-tick it will only execute the last one.

I had to change line 644 `return {walking = false, direction = walking.direction}` because it didn't always get initialized properly, tell me if this is an issue.

I tested it with every save from the script i sent @MortenTobiasNielsen , in single player everything is fine. In multiplayer i sometimes got some factorio crashes which i believe is due to making too many saves. I can't debug in MP, but the stack trace doesn't show any mods. I don't think it is a huge issue.

### CODE REVIEW

**Review by MortenTobiasNielsen — COMMENTED** [review]
I don't understand how we can remove start from the tool - how does c++ or lua know where to start if there are multiple saves?

File: Lua Files/control.lua [hunk 1]
```diff
@@ -602,7 +640,7 @@ local function walk()
 	if compatibility_mode then
 		return {walking = false, direction = defines.direction.north}
 	else
-		return {walking = false, direction = walking.direction}
+		return {walking = false, direction = defines.direction.north}
```
[the remark this answers is not in the document]
Comment (theis999) [reply Lua Files/control.lua:643]: Okay.

**Comment by theis999** [comment]
The c++ part does not know where we start. That is the fundamental idea, the c++ part is completely agnostic about the starting step.
The magic happens in the lua part, where because it remembers the state it was in when saving the game, it can resume from that state. Regardless of which save game you load.

**Comment by MortenTobiasNielsen** [comment]
Would be nice to have something which just works, so the user can just use the save step and not have to worry about anything. I will test it out after work and cross my fingers that it just works. :D

File: Lua Files/control.lua [hunk 2]
```diff
@@ -29,25 +29,63 @@ local input
...
-local keep_x = false
-local keep_y = false
+local pos_pos
+local pos_neg
+local neg_pos
+local neg_neg
+local compatibility_mode
+local keep_x
+local keep_y
 local diagonal = false
```
Comment (MortenTobiasNielsen) [inline Lua Files/control.lua:44]: Should this be uninitialized like this rest?

File: Lua Files/control.lua (hunk 1 above)
[the remark this answers is not in the document]
Comment (MortenTobiasNielsen) [reply Lua Files/control.lua:677]: You need to change this back.

**Review by MortenTobiasNielsen — COMMENTED** [review]
Seems to work pretty well, just have two minor comments.

File: Lua Files/control.lua (hunk 2 above)
Comment (theis999) [reply Lua Files/control.lua]: Likely not.

**Review by MortenTobiasNielsen — APPROVED** [review]

**Pull request merged**